app/bundles/PointBundle/Tests/Unit/Model/PointModelTest.php

```diff
@@ -0,0 +1,142 @@
+        $contactTracker       = $this->createMock(ContactTracker::class);
+        $em                   = $this->createMock(EntityManager::class);
+        $security             = $this->createMock(CorePermissions::class);
+        $dispatcher           = $this->createMock(EventDispatcherInterface::class);
+        $router               = $this->createMock(UrlGeneratorInterface::class);
+        $translator           = $this->createMock(Translator::class);
+        $userHelper           = $this->createMock(UserHelper::class);
+        $logger               = $this->createMock(\Psr\Log\LoggerInterface::class);
+        $coreParametersHelper = $this->createMock(CoreParametersHelper::class);
+        $pointGroupModel      = $this->createMock(PointGroupModel::class);
+
+        // Create a partial mock of the PointModel class
+        $this->pointModel = $this->getMockBuilder(PointModel::class)
+            ->setConstructorArgs([
+                $requestStack,
+                $this->ipLookupHelper,
+                $leadModel,
+                $mauticFactory,
+                $contactTracker,
+                $em,
+                $security,
+                $dispatcher,
+                $router,
+                $translator,
+                $userHelper,
+                $logger,
+                $coreParametersHelper,
+                $pointGroupModel,
+            ])
+            ->onlyMethods(['getRepository'])
+            ->getMock();
+
+        // Configure the getRepository method to return our repository mock
+        $this->pointModel->method('getRepository')
+            ->willReturn($this->pointRepository);
+    }
+
+    public function testTriggerActionReturnsEarlyWhenNoAvailablePoints(): void
+    {
+        // Arrange
+        $type = 'email.send';
+        $lead = new Lead();
+
+        // Configure repository to return empty array for getPublishedByType
+        $this->pointRepository->expects($this->once())
+            ->method('getPublishedByType')
+            ->with($type)
+            ->willReturn([]);
+
+        // IP lookup helper should not be called if we return early
+        $this->ipLookupHelper->expects($this->never())
+            ->method('getIpAddress');
+
+        // Act
+        $this->pointModel->triggerAction($type, null, null, $lead, true);
+
+        // Assert is handled by expectations on mocks - test passes if ipLookupHelper is never called
+    }
+
+    public function testTriggerActionContinuesWhenPointsAreAvailable(): void
+    {
+        // Arrange
+        $type  = 'email.send';
+        $lead  = new Lead();
+        /**
+         * @var Point&\PHPUnit\Framework\MockObject\MockObject
+         */
+        $point = $this->createMock(Point::class);
+
+        // Configure repository to return a point for getPublishedByType
+        $this->pointRepository->expects($this->once())
+            ->method('getPublishedByType')
+            ->with($type)
+            ->willReturn([$point]);
```

Comment:
Let's not use mocks for DTO classes with no PHP service dependencies.
```suggestion
        // Configure repository to return a point for getPublishedByType
        $this->pointRepository->expects($this->once())
            ->method('getPublishedByType')
            ->with($type)
            ->willReturn([new Point()]);
```